### ISSUE TITLE
HAY QUE MODIFICAR * 01 - Regresión Logística

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/najuvgz-intro-ml#2]
@najuvgz 

El EDA es muy corto, echo de menos muchas cosas :(

#### **❌ Lo que falta:**
1. **Análisis de correlaciones**: No analizas relaciones entre variables
2. **Outliers**: No identificas ni graficas valores atípicos
3. **Distribuciones**: No analizas distribuciones de variables numéricas
4. **Feature importance**: No analizas qué variables son más predictivas

 
**También te digo que me ha gustado mucho la nueva variable de _was_previously_contacted_**

2. En el modelo:

#### **❌Errores:**
1.   **Usas Label Encoder para TODAS las variables categóricas**: solo usar para variables ordinarias, ok para education por ejemplo, pero no para job, marital, contact, etc.  Esto podría crear jerarquías falsas. Las variables categóricas sin orden se convierten en numéricas con un orden artificial.

```
# LabelEncoder podría asignar:
'admin.' = 0, 'blue-collar' = 1, 'entrepreneur' = 2
# El modelo interpretará admin < blue-collar < entrepreneur (FALSO)

```

podriamos usar one label encoder para el resto:

```py
#  Usar OneHot para categóricas nominales
from sklearn.preprocessing import OneHotEncoder

nominal_vars = ['job', 'marital', 'contact', 'month', 'day_of_week', 'poutcome']
df_encoded = pd.get_dummies(df_model, columns=nominal_vars)

# Solo LabelEncoder para ordinales
ordinal_vars = ['education', 'default', 'housing', 'loan', 'y']
le = LabelEncoder()
for col in ordinal_vars:
    df_model[col] = le.fit_transform(df_model[col])
```

2. **No usas scaler en variables numéricas** lo que te da el fallo de convergencia. 

```py
# ❌ PROBLEMÁTICO: Sin scaling de variables numéricas
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3)
model = LogisticRegression(max_iter=1000)
model.fit(X_train, y_train)  # ❌ Sin scaling

# ✅ CORRECTO: Aplicar StandardScaler
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)
```
El modelo no converge por escalas distintas, se sesga hacia variables con una mayor escala.

  <img width="956" height="605" alt="Image" src="https://github.com/user-attachments/assets/64f2b25c-1477-467c-b488-9777a62e1f9a" />

3. **No estás evaluando el modelo con sus métricas**: 

Esta parte es muy importante siempre, también si se hacen distintos modelos para compararlos. Es algo que no debería faltar en ningún trabajo.
Deberías incluir:

```py
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, roc_auc_score

accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
f1 = f1_score(y_test, y_pred)
auc = roc_auc_score(y_test, y_pred_proba[:, 1])
```
